### PR TITLE
refactor push context

### PR DESCRIPTION
**Pull request by ramaraochavali**
Refactor PushContext in to multiple sub structs for easy management
Rename few variables in accessbuilder  to match with existing names.

[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

Pull Request Attributes

Please check any characteristics that apply to this pull request. 

[ ] Does not have any changes that may affect Istio users.

### CODE REVIEW

File: pilot/pkg/model/push_context.go [hunk 1]
```diff
@@ -49,6 +49,86 @@ type Metrics interface {
...
+	privateByNamespace map[string][]*Service
+	// public are services reachable within the mesh with exportTo "*"
+	public []*Service
+	// exportedToNamespace are services that were made visible to this namespace
+	// by an exportTo explicitly specifying this namespace.
+	exportedToNamespace map[string][]*Service
+
+	// HostnameAndNamespace has all services, indexed by hostname then namespace.
+	HostnameAndNamespace map[host.Name]map[string]*Service `json:"-"`
+	Hostname             map[host.Name]*Service            `json:"-"`
```
Comment (rshriram) [inline pilot/pkg/model/push_context.go:64]: why is this public?

**Review by rshriram — APPROVED** [review]

**Review by howardjohn — APPROVED** [review]

**Pull request merged**